### PR TITLE
Add an `export` command

### PR DESCRIPTION
**Pull request by edgarrmondragon**
- I refactored some of the code in `cli/packages/cmd/run.go` so it could be
  reused for the new command.
- I added a docs page for the new command.

Closes https://github.com/Infisical/infisical/issues/62

### CODE REVIEW

File: cli/packages/cmd/export.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+// Format environment variables as a CSV file
+func formatAsCSV(envs []models.SingleEnvironmentVariable) string {
+	var csv string
+	csv += "Key,Value\n"
+	for _, env := range envs {
+		csv += fmt.Sprintf("%s,\"%s\"\n", env.Key, env.Value)
+	}
+	return csv
```
Comment (edgarrmondragon) [inline cli/packages/cmd/export.go:115]: This a very naive way of producing a CSV string, but it's probably good enough for a first iteration.

File: cli/packages/cmd/export.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+			substitutions := util.SubstituteSecrets(envsFromApi)
+			output, err = formatEnvs(envName, substitutions, format)
+			if err != nil {
+				log.Errorln(err)
+				return
+			}
+		} else {
+			output, err = formatEnvs(envName, envsFromApi, format)
+			if err != nil {
+				log.Errorln(err)
```
Comment (maidul98) [inline cli/packages/cmd/export.go:78]: We should change this to a user friendly message and add the full message to debug log

File: cli/packages/cmd/export.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+func init() {
+	rootCmd.AddCommand(exportCmd)
+	exportCmd.Flags().StringP("env", "e", "dev", "Set the environment (dev, prod, etc.) from which your secrets should be pulled from")
+	exportCmd.Flags().String("projectId", "", "The project ID from which your secrets should be pulled from")
+	exportCmd.Flags().Bool("expand", true, "Parse shell parameter expansions in your secrets")
+	exportCmd.Flags().StringP("format", "f", "dotenv", "Set the format of the output file (dotenv, json, csv)")
+}
+
+// Format according to the format flag
+func formatEnvs(_ string, envs []models.SingleEnvironmentVariable, format string) (string, error) {
```
Comment (maidul98) [inline cli/packages/cmd/export.go:95]: is there a reason why the parameter `_ string` is here but not used?

File: cli/packages/cmd/export.go (hunk 2 above)
Comment (maidul98) [reply cli/packages/cmd/export.go:78]: Never mind, i see that you set the custom error message in the method

File: cli/packages/cmd/export.go (hunk 1 above)
Comment (maidul98) [reply cli/packages/cmd/export.go:115]: Yeah, i think this is fine for first iteration. If you would like to use the standard csv package from go lang check this out https://pkg.go.dev/encoding/csv#example-Writer

File: docs/cli/export.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+## Description
+
+Export environment variables from the platform into a file format.
+
+## Options
+
+| Option        | Description                                                                                                 | Default value |
+| ------------- | ----------------------------------------------------------------------------------------------------------- | ------------- |
+| `--env`       | Used to set the environment that secrets are pulled from. Accepted values: `dev`, `staging`, `test`, `prod` | `dev`         |
+| `--projectId` | Used to link a local project to the platform (required only if injecting via the service token method)      | `None`        |
```
Comment (maidul98) [inline docs/cli/export.mdx:18]: I think we should change this to `Used to determine from which infisical project your secrets will be exported from (only required if injecting via the service token method).`. What do you think?

File: cli/packages/cmd/export.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+	FormatJson   string = "json"
+	FormatCSV    string = "csv"
+)
+
+// exportCmd represents the export command
+var exportCmd = &cobra.Command{
+	Use:                   "export",
+	Short:                 "Used to export environment variables to a file",
+	DisableFlagsInUseLine: true,
+	Example:               "infisical export --env=prod --format=json",
```
Comment (maidul98) [inline cli/packages/cmd/export.go:28]: I think also adding the example to export to a file will be helpful. Thoughts?

**Comment by maidul98** [comment]
By the way, you should join our slack to communicate easier! https://join.slack.com/t/infisical-users/shared_invite/zt-1kdbk07ro-RtoyEt_9E~fyzGo_xQYP6g

File: cli/packages/cmd/export.go (hunk 3 above)
Comment (edgarrmondragon) [reply cli/packages/cmd/export.go:95]: Yeah, leftover from a refactor when I was planning to use the environment name. I'll remove it

File: docs/cli/export.mdx (hunk 4 above)
Comment (edgarrmondragon) [reply docs/cli/export.mdx:18]: Yeah, makes sense. Bad copy & paste 😅

File: cli/packages/cmd/export.go (hunk 5 above)
Comment (edgarrmondragon) [reply cli/packages/cmd/export.go:28]: Makes sense. I'll add that.

**Pull request merged**